### PR TITLE
Update the turn and continue phrases to include junction names and guide signs

### PR DESCRIPTION
**Pull request by dgearhart**
# Issue

Updated the following phrases to include junction names and guide signs

1. bear
2. bear_verbal
3. continue
4. continue_verbal
5. continue_verbal_alert
6. sharp
7. sharp_verbal
8. turn
9. turn_verbal
10. uturn
11. uturn_verbal

## Example#1 junction name
``` diff
< BEFORE
< 2: Turn right. 
> AFTER
> 2: Turn right at 万年橋東.
```

## Example#2 junction name
``` diff
< BEFORE
< 2: Turn left onto 晴海通り/304/Harumi-dori.
> AFTER
> 2: Turn left at 万年橋東.
```

## Example#3 junction name
``` diff
< BEFORE
< 2: Make a right U-turn at みゆき通り/Miyuki-dori onto 15/中央通り/旧東海道/Chuo-dori Avenue.
> AFTER
> 2: Make a right U-turn at 銀座五丁目.
```

## Example#4 guide signs
``` diff
< BEFORE
< 2: Turn right onto Forstenrieder Straße/St 2344.
<    VERBAL_ALERT: Turn right onto Forstenrieder Straße.
<    VERBAL_PRE: Turn right onto Forstenrieder Straße, St 23 44. Then You will arrive at Forstenrieder Straße.
> AFTER
> 2: Turn right toward A 95/München.
>    VERBAL_ALERT: Turn right toward A 95.
>    VERBAL_PRE: Turn right toward A 95, München. Then You will arrive at Forstenrieder Straße.
```

## Example#5 guide signs
``` diff
< BEFORE
< 2: Bear right.
<    VERBAL_ALERT: Bear right.
<    VERBAL_PRE: Bear right.
> AFTER
> 2: Bear right toward Hersheypark/Arena/Stadium/Giant Center.
>    VERBAL_ALERT: Bear right toward Hersheypark.
>    VERBAL_PRE: Bear right toward Hersheypark, Arena.
```
## Tasklist
 - [x] RAD tests
 - [x] Add gurka tests
 - [x] Add pinpoint tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/odin/directionsbuilder.cc [hunk 1]
```diff
@@ -101,8 +101,6 @@ void DirectionsBuilder::Build(Api& api) {
 
       // Return trip directions
       PopulateDirectionsLeg(options, &etp, maneuvers, trip_directions);
-
-      LOG_INFO("maneuver_count::" + std::to_string(trip_directions.maneuver_size()));
```
Comment (mandeepsandhu) [inline src/odin/directionsbuilder.cc:105]: Why were these removed? Is this because of the excessive logging we've seen recently? Shouldn't that be fixed by changing the log level, instead of removing the log statement?

Comment (dgearhart) [reply src/odin/directionsbuilder.cc:105]: not needed - from old debug sessions

File: src/odin/narrativebuilder.cc [hunk 2]
```diff
@@ -978,14 +1047,30 @@ std::string NarrativeBuilder::FormTurnInstruction(Maneuver& maneuver) {
...
+    // Assign guide sign
+    guide_sign = maneuver.signs().GetGuideString(element_max_count, limit_by_consecutive_count);
+  } else if (maneuver.HasJunctionNameSign()) {
+    // Set the junction phrase - it takes priority over street names
+    phrase_id = 4;
+    // Assign guide sign
+    junction_name =
+        maneuver.signs().GetJunctionNameString(element_max_count, limit_by_consecutive_count);
+  } else {
+    if (!street_names.empty()) {
```
Comment (mandeepsandhu) [inline src/odin/narrativebuilder.cc:1065]: Since only one of these will affect the value set in `phrase_id`, an `if-else` block would read better.

File: src/odin/narrativebuilder.cc [hunk 3]
```diff
@@ -1199,27 +1358,47 @@ std::string NarrativeBuilder::FormVerbalUturnInstruction(Maneuver& maneuver,
...
+    phrase_id = 6;
+    // Assign guide sign
+    junction_name =
+        maneuver.signs().GetJunctionNameString(element_max_count, limit_by_consecutive_count, delim,
+                                               maneuver.verbal_formatter());
+  } else {
+    if (!street_names.empty()) {
       phrase_id += 1;
+      if (maneuver.to_stay_on()) {
+        phrase_id += 1;
```
Comment (mandeepsandhu) [inline src/odin/narrativebuilder.cc:1381]: Why do we do "add" to `phrase_id` in some places, and set it to a specific value in others? Seems like `phrase_id` is an index into a map, so not sure why use the `+=` here and other places.

File: src/odin/narrativebuilder.cc (hunk 2 above)
Comment (dgearhart) [reply src/odin/narrativebuilder.cc:1065]: all three need to be in the `else` block

File: src/odin/narrativebuilder.cc (hunk 3 above)
Comment (dgearhart) [reply src/odin/narrativebuilder.cc:1381]: we add when a certain attribute can map to multiple phrase values depending on other attributes - in this case a non empty street name can map to 1, 2, 4, or 5

File: src/odin/narrativebuilder.cc (hunk 2 above)
Comment (mandeepsandhu) [reply src/odin/narrativebuilder.cc:1065]: Yeah, I was talking about the 3 `if`s in the `else` block. It looks like we only set `phrase_id` if one of the conditions is true:
```c++
if (maneuver.to_stay_on()) {
   phrase_id = 3;
} else if (!begin_street_names.empty()) {
   phrase_id = 2;
} else if (!street_names.empty()) {
   phrase_id = 1;
}
```

File: src/odin/narrativebuilder.cc (hunk 3 above)
Comment (mandeepsandhu) [reply src/odin/narrativebuilder.cc:1381]: Ok, thanks for the clarification.

File: src/odin/narrativebuilder.cc (hunk 2 above)
Comment (dgearhart) [reply src/odin/narrativebuilder.cc]: I cleaned up - thanks @mandeepsandhu

File: src/odin/narrativebuilder.cc [hunk 4]
```diff
@@ -1057,14 +1146,32 @@ std::string NarrativeBuilder::FormVerbalTurnInstruction(Maneuver& maneuver,
...
-  if (!begin_street_names.empty()) {
-    phrase_id = 2;
-  }
-  if (maneuver.to_stay_on()) {
-    phrase_id = 3;
+  std::string junction_name;
+  std::string guide_sign;
+
+  if (maneuver.HasGuideSign()) {
+    // Set the toward phrase - it takes priority over street names and junction name
```
Comment (danpaz) [inline src/odin/narrativebuilder.cc:1153]: Just noting we may want to revisit the priority ordering of GuideSign > JunctionName > StreetName at some point. Also it may be preferable to announce JunctionName and GuideSign together `Turn left at X toward Y`. No change needed for this PR.

Comment (dgearhart) [reply src/odin/narrativebuilder.cc:1153]: I was thinking the same possibility but kept to the current plan for now
